addons/calendar_sms/tests/test_calendar_sms.py

```diff
@@ -31,3 +83,20 @@ def test_attendees_with_number(self):
             'partner_ids': [(6, 0, [self.partner_phone.id, self.partner_no_phone.id])],
         })._sms_get_default_partners()
         self.assertEqual(len(attendees), 1, "There should be only one partner retrieved")
+
+    def test_send_reminder_match_both_events(self):
+        """
+        Test that only the necessary SMS messages are sent,
+         with each SMS template correctly matching its corresponding alarm duration and the event.
+        """
+        with self.mockSMSGateway():
+            lastcall = fields.Datetime.now() - timedelta(hours=1)
+            self.env['calendar.alarm_manager'].with_context(lastcall=lastcall)._send_reminder()
+
+            self.assertEqual(len(self._sms), 2)
+            self.assertSMS(self.partner_phone, self.partner_phone.phone, 'sent',
+                           content=self.sms_template_1h.body)
+            self.assertSMS(self.partner_phone_2, self.partner_phone_2.phone, 'sent',
+                           content=self.sms_template_24h.body)
```

Comment:
Also just fyi, you can call these outside of the `with` block, _sms remains set until the next mock call